src/vehicle.cpp

```diff
@@ -2787,27 +2784,33 @@ std::vector<int> vehicle::boarded_parts() const
     std::vector<int> res;
     for (size_t p = 0; p < parts.size(); p++) {
         if (part_flag (p, VPFLAG_BOARDABLE) &&
-                parts[p].has_flag(vehicle_part::passenger_flag)) {
+            get_passenger (p)) {
             res.push_back ((int)p);
         }
     }
     return res;
 }
 
-player *vehicle::get_passenger(int p) const
+template<typename T>
+T *vehicle::get_passenger(int p) const
 {
     p = part_with_feature (p, VPFLAG_BOARDABLE, false);
-    if (p >= 0 && parts[p].has_flag(vehicle_part::passenger_flag))
+    if (p >= 0)
     {
-     const int player_id = parts[p].passenger_id;
-     if( player_id == g->u.getID()) {
-      return &g->u;
-     }
-        return g->npc_by_id( player_id );
+	T *psg = g->critter_at<T> (global_part_pos3(p));
+	if (psg && psg->in_vehicle()) {
```

Comment:
This `in_vehicle` checks looks a bit redundant. All it does is: ensure there is a vehicle with a part that has the boardable flag. But that has already been done in this function. So what's the point? In other words: when will this ever return `false`?